build/windows/omsupply_server.suf

```diff
@@ -4186,10 +4195,34 @@ if not hasOmsupplyInstalled then
 	config = config .. "  max_file_size: 10\r\n";
 	TextFile.WriteFromString(SessionVar.Expand("%AppFolder%\\configuration\\local.yaml"), config, false);
 
-	-- Create omSupply Server keys in registry
-	Registry.CreateKey(HKEY_LOCAL_MACHINE, "Software\\omSupply Server");
-	Registry.SetValue(HKEY_LOCAL_MACHINE, "Software\\omSupply Server", "Path", omSupplyServerPath, REG_SZ);
-	Registry.SetValue(HKEY_LOCAL_MACHINE, "Software\\omSupply Server", "Database", omSupplyServerDatabase, REG_SZ);
+	-- Create environment variable for postgres libraries required for diesel orm (backend)
+	if (omSupplyServerDatabase == "postgres") then
+		local envKey = "System\\CurrentControlSet\\Control\\Session Manager\\Environment";
+		local path = Registry.GetValue(HKEY_LOCAL_MACHINE, envKey, "Path", true);
+		local binPath = "C:\\Program Files\\PostgreSQL\\14\\bin";

```

Comment:
We can't guarantee that it will be postgres v14 installed, so this isn't safe to do.
Could you read the key `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\PostgreSQL\Installations` and get the path from there instead please.

Something like this is needed I think:

```
local pgInstalls = Registry.GetKeyNames (HKEY_LOCAL_MACHINE, "SOFTWARE\\PostgreSQL\\Installations");
if Table.Count(pgInstalls) > 0 then
  -- Does the GetKeyNames return full path? or do we need to concatenate?
  local basePath = Registry.GetValue(HKEY_LOCAL_MACHINE, pgInstalls[1] .. "\\Base Directory", "Path", true);
  local binPath = basePath .. "\\bin";
...etc
end
```

Reply:
Oh, the idea behind 14 was that version is being used as a suitable postgres version for the omsupply in the installer so used it but having dynamic check make sense!